src/openms/source/TRANSFORMATIONS/FEATUREFINDER/MultiplexFiltering.cpp

```diff
@@ -93,9 +95,11 @@ namespace OpenMS
         mz_shifts_actual.push_back(std::numeric_limits<double>::quiet_NaN());
         mz_shifts_actual_indices.push_back(-1);
       }
-
     }
 
+    // early-out: not peaks found 
+    if (found_peaks < peaks_per_peptide_min_) return -1;
```

Comment:
`peaks_per_peptide_min_ * pattern.getMassShiftsCount()`

`found_peaks_` counts all m/z shifts. `peaks_per_peptide_min_` refers to a single peptide. `pattern.getMassShiftsCount()` is the number of peptides, e.g. 3 for a triplet search.